src/WorkflowCore/Services/WorkflowExecutor.cs

```diff
@@ -242,18 +250,14 @@ private async Task DetermineNextExecutionTime(WorkflowInstance workflow, Workflo
             }
 
             if ((workflow.NextExecution != null) || (workflow.ExecutionPointers.Any(x => x.EndTime == null)))
+            {
                 return;
+            }
 
             workflow.Status = WorkflowStatus.Complete;
             workflow.CompleteTime = _datetimeProvider.UtcNow;
 
-            using (var scope = _serviceProvider.CreateScope())
```

Comment:
Will moving this break existing expectations on when this fires? ie. this could break existing use cases out there?

Reply:
@glucaci ^

Reply:
Somehow wrote a comment yesterday and is not here anymore...
My opinion is that this is not a breaking change because everything that is public to the client is the `workflow` itself and the `definition`, so is nothing that we changed in the public API.

Also moving the position where the `scope` gets created is not a breaking change.

Can you detailed what did you think that is creating a breaking change?

Reply:
What about consumers of workflow-core that are using the post middleware and only expecting it to run on workflow completion. Now it is going to run every time a step executes, multiple times throughout the workflow's life.

Reply:
Yes, makes sens, I read again the docu: ` get run after the workflow has completed`.
In this case we can create another one and call it like above.

Do you have any naming preference?
Proposal: `WorkflowMiddlewarePhase.OnChange`